DXMainClient/Domain/MainClientConstants.cs

```diff
@@ -20,8 +27,54 @@ public static class MainClientConstants
 
         public static OSVersion OSId = OSVersion.UNKNOWN;
 
+        private static Action<string, string, bool> displayErrorAction = null;
+        /// <summary>
+        /// Gets or sets the action to perform to notify the user of an error.
+        /// </summary>
+        public static Action<string, string, bool> DisplayErrorAction
+        {
+            get => displayErrorAction ??= InitialDisplayErrorAction;
+            set => displayErrorAction = value;
+        }
+
+        public static Action<string, string, bool> InitialDisplayErrorAction = (title, error, exit) =>
+        {
+            Console.WriteLine(title);
+            Console.WriteLine();
+            Console.WriteLine(error);
+#if WINFORMS
+            MessageBox.Show(error, title, MessageBoxButtons.OK, MessageBoxIcon.Error);
+#else
+            string tempfile = SafePath.CombineFilePath(Path.GetTempPath(), "xna-cncnet-client-error.txt");
+            using (StreamWriter writer = new StreamWriter(tempfile))
+            {
+                writer.WriteLine(title);
+                writer.WriteLine();
+                writer.WriteLine(error);
+            }
+            ProcessLauncher.StartShellProcess(tempfile);
+#endif
+            if (exit)
+                Environment.Exit(1);
+        };
+
+        public static Action<string, string, bool> DefaultDisplayErrorAction = (title, error, exit) =>
+        {
+            Logger.Log(FormattableString.Invariant($"{(title is null ? null : title + Environment.NewLine + Environment.NewLine)}{error}"));
+#if WINFORMS
+            MessageBox.Show(error, title, MessageBoxButtons.OK, MessageBoxIcon.Error);
+#else
+            ProcessLauncher.StartShellProcess(ProgramConstants.LogFileName);
+#endif
+
+            if (exit)
+                Environment.Exit(1);
+        };
```

Comment:
Kind of don't understand the distinction and duplication of the funcs here (also if the func is not changed it may be put as a regular static method), would need an explanation (could be latter put in some docstrings/comments if everything is ok).

Reply:
Should I reduce these two error actions to one, by determining if the logger has been initialized yet?

Reply:
Was about to suggest that, it seems more appropriate to just check if the logger is inited inside one action.

Reply:
Updated